### ISSUE TITLE
aws-deployment-framework-base-templates CodeBuild project timeout

### ISSUE THREAD

**Issue by Nr18** [awslabs/aws-deployment-framework#431]
## Description

We use a lot of regions and have a lot of accounts. When performing an upgrade you might not have enough time to perform all the necessary steps. For this reason we updated the timeout of the `aws-deployment-framework-base-templates` project from `40` to `240`. But after an hour the temporary credentials expire and the project stops.

## Workaround

Trigger the pipeline again by clicking the "Release" button.

## Solution(s)

There are 2 solutions possible:

* When the credentials almost expire they need to be renewed.
* Replace the whole process with StepFunctions (preferred option as it also includes cost savings)

I know that there are some initiatives to move to StepFunctions already. This is just another reason why it's a good idea.

**Comment by AndreasAugustin** [comment]
Maybe related to #371

**Comment by Nr18** [comment]
@AndreasAugustin not quite, I encountered the timeout on a CodeBuild project. #371 is about Lambda timeouts.

But you do raised a concern in ADF itself, there is no proper retry logic on timeouts.
